Enable scheduled publishing for imports

kontrollanten opened the pull request:
## Description
* Enable scheduled publishing for import URL/torrent.
* Disable instead of hide the schedule publishing option in cases where it's not enabled.

## Reflections
* Shouldn't it be possible to schedule the publishing even if the video is marked as internal or unlisted?

## Related issues
#3134 

## Has this been tested?
- [x] 👍 yes, light tests as follows are enough

1) Import a URL
1) Check that it's possible to schedule the publishing.

1) Import a torrent
1) Check that it's possible to schedule the publishing.

1) Set a videos privacy setting to _Internal_
1) Edit the video and see that _Scheduled_ is disabled.

## Screenshots

<!-- delete if not relevant -->
![image](https://user-images.githubusercontent.com/6680299/101933872-1325fd00-3bdd-11eb-8da6-56635d29577e.png)

kontrollanten commented:
I just found out that I didn't add the schedule option in the first step of import. What's the reason that the privacy and channel fields are both in step 1 and 2 of the import process? Can we remove the fields in the first step to keep it simpler?

Chocobozzz commented:
Hello,

We disabled the schedule privacy from import because we're not sure when the file will be imported, so the feature could not work as expected. And we did not have too many users asking this feature for imports. I think most of them will import the file in private, and schedule a privacy change afterwards.

kontrollanten commented:
I see. What about just disable the option instead of hide it (so the user at least knows that this functionality exists)? Shall I remove the privacy option in import step 1 (since it's redundant)?

Chocobozzz commented:
> I see. What about just disable the option instead of hide it (so the user at least knows that this functionality exists)?

I don't think we should display a disabled thing that cannot be enabled in this screen. It would just be confusing for users.

> Shall I remove the privacy option in import step 1 (since it's redundant)?

We added the channel and privacy in the first step to highlight these settings, even if you can update them in the second screen

kontrollanten commented:
> > I see. What about just disable the option instead of hide it (so the user at least knows that this functionality exists)?
> 
> I don't think we should display a disabled thing that cannot be enabled in this screen. It would just be confusing for users.

I see your point, but I don't agree. It cannot be enabled in this screen, but on this video. Me myself thought PeerTube lacked the functionality of scheduling publishing during my first month using it, since I just imported videos and never saw that alternative (and since there's an issue for this I'm not alone about that). Can we add a question mark with a popover that describes the different privacy options (and why _Scheduled_ is disabled)?


> > Shall I remove the privacy option in import step 1 (since it's redundant)?
> 
> We added the channel and privacy in the first step to highlight these settings, even if you can update them in the second screen

Sorry maybe I'm missing something, but I don't see why it should be highlighted. It'd make sense if it'd be a quick way to just skip the second step, but the user has to go through the second step and see those fields again anyway.

Chocobozzz commented:
> I see your point, but I don't agree. It cannot be enabled in this screen, but on this video. Me myself thought PeerTube lacked the functionality of scheduling publishing during my first month using it, since I just imported videos and never saw that alternative (and since there's an issue for this I'm not alone about that). 

Okay then.

>  Can we add a question mark with a popover that describes the different privacy options (and why Scheduled is disabled)?

Adding in the schedule description "this option will be can be enabled after the import" could be interesting

>  but I don't see why it should be highlighted.

The multi channel concept is quite different from other video platforms so we want the user to see it can upload the video in a particular channel. The privacy option is also essential, so the user understands at the first step it will upload this video in this particular privacy.

kontrollanten commented:
I've updated with the following changes:
* I reverted the commit which made it possible to schedule an import.
* Privacy options are now shown as a helper (which makes it possible to read the whole description for the alternatives).
* The schedule option is always shown, but disabled on those places where it previously wasn't shown.
* A short description is shown why it's disabled (e.g. "only available after import")

Maybe I did a more complex solution than necessary (it could have been done with just writing a general description beside the Privacy label), but I think it's a good step forward anyway.

Upload/import step 1:
![upload-import-step-1](https://user-images.githubusercontent.com/6680299/103894779-b2141b00-50ef-11eb-8ffe-cbeb81e9bead.png)

Import step 2:
![import-step2](https://user-images.githubusercontent.com/6680299/103894821-be987380-50ef-11eb-82dc-3e4659bbcfdd.png)
![import-step2-options](https://user-images.githubusercontent.com/6680299/103894835-c5bf8180-50ef-11eb-91b8-b485fa6cba0f.png)

Upload step 2:
![upload-step2](https://user-images.githubusercontent.com/6680299/103894869-ce17bc80-50ef-11eb-9688-2752b4bc1b58.png)

Edit a public video:
![edit-public-video](https://user-images.githubusercontent.com/6680299/103894755-a9bbe000-50ef-11eb-9355-0a518f27be1b.png)

Edit a private video:
![edit-pivate-video](https://user-images.githubusercontent.com/6680299/103894725-a0cb0e80-50ef-11eb-8683-76276d1c6702.png)

Chocobozzz commented on shared/models/videos/video-privacy.enum.ts:
```diff
@@ -1,4 +1,5 @@
 export const enum VideoPrivacy {
+  SCHEDULED = -1,
```
This is not a video privacy, then should not be added to this enum. It's the reason why we used `VideoEdit.SPECIAL_SCHEDULED_PRIVACY`

kontrollanten replied:
True, but the form field name is "privacy". Can we rename the enum to `VideoPrivacyOption` or `PublishOption`?

Chocobozzz replied:
This enum? No because it's used by the server, that only accept these values. But you can create another enum in the client named `PublishOption`.

kontrollanten commented:
I'll close this since I don't have time to finish it.
